src/Symfony/Bundle/FrameworkBundle/KernelBrowser.php

```diff
@@ -56,7 +56,7 @@ public function getKernel(): KernelInterface
      */
     public function getProfile(): HttpProfile|false|null
     {
-        if (null === $this->response || !$this->getContainer()->has('profiler')) {
```

Comment:
the right fix is to check for an uninitialized property, to avoid a TypeError if calling this method before doing a request (this was forgotten when migrating to a typed property)

Reply:
Updated as well with `isset()`

Reply:
see also #58626